opsimulator/opsimulator.go

```diff
@@ -468,6 +475,22 @@ func (opSim *OpSimulator) checkInteropInvariants(ctx context.Context, logs []typ
 				return fmt.Errorf("executing and initiating message fields are not equal")
 			}
 
+			interopStart, err := opSim.crossL2Inbox.InteropStart(&bind.CallOpts{})
+			if err != nil {
+				return fmt.Errorf("failed to fetch interop start: %w", err)
+			}
+			if interopStart.Cmp(identifier.Timestamp) > 0 {
+				return fmt.Errorf("interop start is greater than initiating message timestamp")
+			}
+
+			isInDependencySet, err := opSim.l1BlockInterop.IsInDependencySet(&bind.CallOpts{}, big.NewInt(identifier.ChainId.Int64()))
+			if err != nil {
+				return fmt.Errorf("failed to check if chain id is in dependency set: %w", err)
+			}
+			if !isInDependencySet {
+				return fmt.Errorf("chain id is not in dependency set")
+			}
+
```

Comment:
now that these checks have been from `CrossL2Inbox`, it is expected that they take place off chain, so I am adding them here in opsimulator. [PR](https://github.com/ethereum-optimism/optimism/pull/13626/files#diff-c899f14eedc7928371b7e497dd12658c2452d9a171856b2b001cc20885fe6dc8L192) that removed the checks from `CrossL2Inbox`. Also, see this issue for more details: https://github.com/ethereum-optimism/optimism/issues/13658